src/keymap-manager.coffee

```diff
@@ -409,15 +410,84 @@ class KeymapManager
   # target is `.defaultTarget` if that property is assigned on the keymap.
   #
   # * `event` A `KeyboardEvent` of type 'keydown'
-  handleKeyboardEvent: (event) ->
+  handleKeyboardEvent: (event, {replay, disabledBindings}={}) ->
+    # Handling keyboard events is complicated and very nuanced. The
+    # complexity is all because of multi-stroke bindings. For example:
+    #
+    # 'ctrl-a b c': 'my-sweet-command' // This is a binding
+    #
+    # This example means the user can type `ctrl-a` then `b` then `c`, and after
+    # all of those keys are typed, it will dispatch the `my-sweet-command`
+    # command.
+    #
+    # The KeymapManager has a couple member variables to deal with multi-stroke
+    # bindings: `@queuedKeystrokes` and `@queuedKeyboardEvents`. They keep track
+    # of the keystrokes the user has typed. When populated, the state variables
+    # look something like:
+    #
+    # @queuedKeystrokes = ['ctrl-a', 'b', 'c']
+    # @queuedKeyboardEvents = [KeyboardEvent, KeyboardEvent, KeyboardEvent]
+    #
+    # Basically, this `handleKeyboardEvent` function will try to exactly match
+    # the user's keystrokes to a binding. If it cant match exactly, it looks for
+    # partial matches. A partial match might be something like:
+    #
+    # // The binding: 'ctrl-a b c': 'my-sweet-command'
+    # @queuedKeystrokes = ['ctrl-a', 'b'] // The user's keystrokes
+    #
+    # When it finds partially matching bindings, it will set the KeymapManager
+    # into a pending state via `enterPendingState`.
+    #
+    # If a keystroke comes in that either matches a binding exactly, or yields
+    # no partial matches, we will reset the state variables and exit pending
+    # mode. If the keystroke yields no partial matches we will call
+    # `terminatePendingState`
+    #
+    # // Both of these will exit pending state for: 'ctrl-a b c': 'my-sweet-command'
+    # @queuedKeystrokes = ['ctrl-a', 'b', 'c'] // Exact match! Just clear the state variables. Easy.
+    # @queuedKeystrokes = ['ctrl-a', 'b', 'd'] // No hope of matching, terminatePendingState(). Dragons.
+    #
+    # `terminatePendingState` is where things get crazy. Let's pretend the user
+    # typed `['ctrl-a', 'b', 'd']`. There are were no exact matches from this
```

Comment:
I don't think you should mention the example sequence until the example ambiguous bindings are mentioned.